PIMTool.Services/Controllers/ProjectController.cs

```diff
@@ -1,42 +1,80 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Net;
-using System.Net.Http;
+﻿using System.Collections.Generic;
 using System.Web.Http;
-using log4net;
-using Ninject;
+using AutoMapper;
 using PIMTool.Common;
-using PIMTool.Common.BusinessObjects;
+using PIMTool.Services.Resource;
 using PIMTool.Services.Service;
+using PIMTool.Services.Service.Entities;
 
 namespace PIMTool.Services.Controllers
 {
     [RoutePrefix(RouteConstants.ProjectApi)]
     public class ProjectController : ApiController
     {
         private readonly IProjectService _projectService;
-
+        private readonly IMapper _mapper;
         public ProjectController(
-            IProjectService projectService)
+            IProjectService projectService, IMapper mapper)
         {
             _projectService = projectService;
+            _mapper = mapper;
         }
 
         [Route(RouteConstants.GetAllProjects)]
         [HttpGet]
-        public IHttpActionResult GetProjects()
+        public IEnumerable<ProjectResource> GetProjects()
         {
             //[TODO] call real service
-            return Ok(new Project {Name = "Test", Id = 1});
+            var projects = _projectService.GetAll();
+            var resources = _mapper.Map<IEnumerable<ProjectEntity>, IEnumerable<ProjectResource>>(projects);
+            return resources;
         }
 
         [Route(RouteConstants.GetProjectServer)]
         [HttpGet]
-        public IHttpActionResult GetProject(int projectId)
+        public ProjectEntity GetProject(int projectId)
         {
             //[TODO] call real service
-            return Ok(new Project { Name = "Test2", Id = 2 });
+            return _projectService.GetById(projectId);
+        }
+
+        [Route(RouteConstants.DeleteProject)]
+        [HttpDelete]
+        public IHttpActionResult DeleteProject(int projectNumber)
+        {
+            var result = _projectService.DeleteByProjectNumber(projectNumber);
+
+            if (!result.Success)
+            {
+                return BadRequest(result.Message);  
```

Comment:
We should not return the BadRequest. Next step in advantage task please also implement https://weblogs.asp.net/fredriknormen/asp-net-web-api-exception-handling